README.md

```diff
@@ -48,95 +53,118 @@ defmodule MyProject.Mixfile do
+provides a [supervisor] which can be added to your application's supervision
+tree. It's important to add it to your supervision tree with type `:supervisor`
+and not `:worker`.
+
+```elixir
+defmodule Calculator.Application
+  alias Calculator.Generated.Service.Binary.Framed.Server
+
+  def start(_type, _args) do
+    children = [
+      server_child_spec(9090)
+    ]
 
-The server defines a Supervisor, which can be added to your application's supervision tree. When adding the server to your applications supervision tree, use the `supervisor` function rather than the `worker` function.
+    opts = [strategy: :one_for_one, name: Calculator.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
 
-## Using the binary protocol directly
+  defp server_child_spec(port) do
+    %{
+      id: Server,
+      start: {Server, :start_link, [Calculator.ServiceHandler, port]},
+      type: :supervisor
+    }
+  end
+end
+```
 
-Each thrift struct, union and exception also has a `BinaryProtocol` module generated for
-it. This module lets you serialize and deserialize its own type easily.
+[behaviour]: https://elixir-lang.org/getting-started/typespecs-and-behaviours.html#behaviours
+[supervisor]: https://elixir-lang.org/getting-started/mix-otp/supervisor-and-application.html
 
-For example:
+## Serialization
+
+A `BinaryProtocol` module is generated for each Thrift struct, union, and
+exception type. You can use this interface to easily serialize and deserialize
+your own types.
 
 ```elixir
-iex(1)> serialized = %User{username: "stinky" id: 1234, first_name: "Stinky", last_name: "Stinkman"}
-|> User.BinaryProtocol.serialize
+iex> alias Calculator.Generated.Vector
+iex> data = %Vector{x: 1, y: 2, z: 3}
+|> Vector.BinaryProtocol.serialize
 |> IO.iodata_to_binary
-iex(2)> User.BinaryProtocol.deserialize(serialized)
-{%User{username: "stinky" id: 1234, first_name: "Stinky", last_name: "Stinkman"}, ""}
+iex> Vector.BinaryProtocol.deserialize(data)
+{%Calculator.Generated.Vector{x: 1.0, y: 2.0, z: 3.0}, ""}
 ```
 
-The return value of the `serialize` function is an [iodata]. You can pass it through `IO.iodata_to_binary` to convert it to a binary. You also can write the iodata directly to a file or socket without converting it.
+## Thrift IDL Parsing
 
-## Other Features
-### Thrift IDL Parsing
+The `Thrift.Parser` module parses [Thrift IDL][idl] documents and produces an
+abstract syntax tree.
 
-This package also contains support for parsing [Thrift IDL][idl]
-files. It is built on a low-level Erlang lexer and parser:
+```elixir
+Thrift.Parser.parse("enum Colors { RED, GREEN, BLUE }")
+%Thrift.AST.Schema{constants: %{},
+ enums: %{Colors: %Thrift.AST.TEnum{name: :Colors,
+    values: [RED: 1, GREEN: 2, BLUE: 3]}}, exceptions: %{}, includes: [],
+ namespaces: %{}, services: %{}, structs: %{}, thrift_namespace: nil,
+ typedefs: %{}, unions: %{}}
+```
+
+There is also a low-level lexer (`:thrift_lexer`) and parser (`:thrift_parser`)
```

Comment:
Do we want to document these? I dont know if we want to be forced to support the intermediate tokens?

Reply:
They've been there for a long time, but I do think it makes sense to remove them.